src/listener/can_open_listener.py

```diff
@@ -0,0 +1,123 @@
+import canopen
+import logging
+from threading import Thread
+
+from listener.listener import Listener
+from canopen.sdo.exceptions import SdoCommunicationError
+from time import strftime, localtime
+
+
+class CanOpenListener(Listener):
+    ''' This class is being used to listen on a network and mainly
+        tries to push the changes around other nodes to it's interpreter.
+    '''
+    def __init__(self, config, interpreter=None):
+        ''' config : yaml_file_stream
+                The configured yaml file that contains details for the
+                connection type.
+            interpreter : CanOpenInterpreter
+                The wanted interpreter that is going to be used.
+        '''
+        super().__init__(config)
+        self.observers = []
+        self.network = self.connect_to_network()
+        self._add_nodes(self.config['canopen']['nodes'])
+        self.interpreter = interpreter
+        if(self.config['canopen']['raw_can_data_logging']):
+            # Creates new network connection because CANopen library has
+            # difficulties with multthreading on a single network connection.
+            self.raw_log_network = self.connect_to_network()
+            Thread(target=self._log_raw_data).start()
```

Comment:
is de logger van python-can zelf niet nuttig hiervoor?
https://python-can.readthedocs.io/en/master/scripts.html#can-logger

Reply:
ik zal er naar kijken

Reply:
We kunnen hiervoor een issue aanmaken en er later naar kijken dit valt buiten de scope